core/primitives/src/views.rs

```diff
@@ -1366,6 +1369,18 @@ pub enum ActionView {
     UseGlobalContractByAccountId {
         account_id: AccountId,
     } = 12,
+    DeterministicStateInit {
```

Comment:
I suggest having single variant with code represented by enum with some user-friendly serialization. I guess [untagged](https://serde.rs/enum-representations.html#untagged) would be the best here.
Something like
```
#[serde(untagged)]
enum GlobalContractIdentifierView {
    CodeHash { hash: CryptoHash },
    AccountId  { account_id: AccountId },
} 
...
DeterministicStateInit {
   code: GlobalContractIdentifierView,
   ...
}
```
then this variant will be represented in json as `{ "code": { "hash": "..." }, ... }` or `{ "code": { "account_id": "..." }, ... }`

in hindsight I should have done the same for global contract related actions.

Reply:
Note: the same can be achieved by using `#[serde(rename = "...")]` attribute on enum variants, while it keeps the code compact:
```rust
#[derive(Serialize)]
#[serde(rename_all = "snake_case")]
enum GlobalContractIdentifierView {
    #[serde(rename = "hash")]
    CodeHash(CryptoHash),
    AccountId(AccountId),
} 

#[derive(Serialize)]
struct DeterministicStateInit {
   code: GlobalContractIdentifierView,
   /// ...
}
```

Reply:
Yes, let's not change `UseGlobalContract` for now